main.cpp

```diff
@@ -1720,13 +1720,33 @@ void load_command::execute(device_map &devices) {
     auto con = get_single_usb_boot_device(devices);
     picoboot_memory_access raw_access(con);
     auto ranges = get_colaesced_ranges(file_access);
+    bool uses_flash = false;
+    uint32_t flash_min = std::numeric_limits<uint32_t>::max();
+    uint32_t flash_max = std::numeric_limits<uint32_t>::min();
     for (auto mem_range : ranges) {
         enum memory_type t1 = get_memory_type(mem_range.from);
         enum memory_type t2 = get_memory_type(mem_range.to);
         if (t1 != t2 || t1 == invalid || t1 == rom) {
             fail(ERROR_FORMAT, "File to load contained an invalid memory range 0x%08x-0x%08x", mem_range.from,
                  mem_range.to);
         }
+        if (t1 == flash) {
+            uses_flash = true;
+            flash_min = std::min(flash_min, mem_range.from);
+            flash_max = std::max(flash_max, mem_range.to);
+        }
+    }
+    if (uses_flash) {
+        int32_t size_guess = guess_flash_size(raw_access);
+        if (size_guess > 0) {
+            assert(flash_max > flash_min);
+            uint32_t flash_data_size = flash_max - flash_min;
+            assert(flash_min >= FLASH_START);
+            uint32_t flash_start_offset = flash_min - FLASH_START;
+            if ((flash_start_offset + flash_data_size) > size_guess) {
```

Comment:
seems like we could just track flash_max (flash_min provides no actual value)

Reply:
I assumed that the `assert(flash_min >= FLASH_START);` check might be useful? Or should I take that out too?